src/main/java/htmlpublisher/HtmlPublisher.java

```diff
@@ -237,6 +238,7 @@ public static boolean publishReports(Run<?, ?> build, FilePath workspace, TaskLi
         for (int i=0; i < reportTargets.size(); i++) {
             // Create an array of lines we will eventually write out, initially the header.
             List<String> reportLines = new ArrayList<>(headerLines);
+            reportLines.add("<script type=\"text/javascript\" src=\"" + Jenkins.get().getRootUrl() + Functions.getResourcePath() + "/plugin/htmlpublisher/js/htmlpublisher.js\"></script>");
```

Comment:
What I need here is the `resURL` variable that's normally available in jelly views. Here's how it's initialized in core: https://github.com/jenkinsci/jenkins/blob/be3d711d690d5cf670ced78019717b4c609eda1b/core/src/main/java/hudson/Functions.java#L305. I cannot do it the same way it's done there because we're not in context of a request here, so `Stapler.getCurrentRequest()` gives me `null`. `Jenkins.get().getRootUrl()` returns URL with slash in the end, so in ends up containing two slashes: `http://localhost:8080/jenkins//static/deadbeef/plugin/...`. Would appreciate a hint if someone know how to do this better.

Reply:
Looking at various usages of `getRootUrl` in the ecosystem, I see a few patterns, such as wrapping the result in URI or URL and invoking `getPath()`, or stuff like this:

```java
    private String getRootUrl() {
        Jenkins jenkins = Jenkins.get();
        String url = isFromRequest() ? jenkins.getRootUrlFromRequest() : jenkins.getRootUrl();
        return StringUtils.stripEnd(url, "/");
    }
```

Reply:
Another common pattern I am seeing is checking if `enkins.get().getRootUrl()` is null or empty, so we should likely handle that case here as well.

Reply:
Applied both suggestions, thanks. Checking for `isFromRequest()` is redundant, `jenkins.getRootUrl()` takes care of that under the hood.